Moise1/Free_Mentors
# #168276633 - Applying my LFA's feedback on UI and Back-end

#### What does this PR do? 
- Displaying a table of users and their request messages 
- Separating the mentee's dashboard from the mentor's 
- Excluding everything from the returned sign up response except the user's token 
- Excluding the user's token, is_mentor, is_admin from the response where necessary. 
- Validate password to have one digit, one capital letter, one small letter, and one special character; all making a minimum of 8 characters.  
 
#### Description of Task to be completed?

#### _Have the following working_ :
 - Write reusable code. 

#### How should this be manually tested?(back-end)

- Clone this GitHub repo on your computer
- Run npm install
- Install [Postman](https://www.getpostman.com/) 
- Start testing the endpoints with Postman 

#### How should this be manually tested?(UI)

- Clone this GitHub repo on your computer
- cd Free_Mentors
- run live-server if you have node new version ready. 
- check all the mentioned features for UI.

#### Any background context you want to provide?
None 
#### What are the relevant pivotal tracker stories?

[#168276633](https://www.pivotaltracker.com/n/projects/2379572)
![frontpage](https://user-images.githubusercontent.com/25052255/64243462-ad41c480-cf07-11e9-9539-c2728e09707b.png)

![mentor_board](https://user-images.githubusercontent.com/25052255/64243110-18d76200-cf07-11e9-906b-40cd15330ff1.png)
![postman](https://user-images.githubusercontent.com/25052255/64243112-196ff880-cf07-11e9-90e2-d79020599e92.png)
